### DISCUSSION TITLE
Understanding Negation - QA for "Leveraging Affirmative Interpretations from Negation Improves Natural Language Understanding"

### DISCUSSION THREAD

**Discussion by eduardoblanco** [eduardoblanco/csc696H-Spring24#2] [General]
See [Understanding Negation](https://github.com/eduardoblanco/csc696H-Spring24/wiki/Understanding-Negation)

This QA should happen before and after the discussion class in general.

I am reusing the materials from the Discussion Lead. I selected the ones that I believe are most interesting.

Please format answers with a numeric reference and then your answer. You do not need to answer all questions to post.

Questions to be discussed:

1. Section 3
   1. Why did the authors use Norwegian and Spanish? What are the requirements to use another language?
   1. Better machine translation (potentially) means better results of the collection process. Is it worth exploring whether this is true?
   1. How much worse is the paper if we get rid of Section 3.4?
   1. What would happen if the authors reported that in 75% of their "(sentence with negation, affirmative interpretation) pairs" the affirmative interpretation contains negation (and Section 4 stays the same)?

1. Section 4
   1. Would it be interesting to investigate if the authors need 153k pairs to train the affirmative interpretation generator? Would it make sense to attempt to collect more pairs?

1. Section 4.1
   1. What do you trust more, the automatic metrics or the manual validation?
   1. Where do the meaning of the scores come from? Do we know who did the manual validation?

1. Section 4.2
   1. Why three natural language inference corpora? Can't they just use one?
   1. Do you need to replicate previous work (== get their results yourself) to add them to your table of results? Can you just copy and paste from their papers?
   1. Why doesn't the second block of Table 5 have results with BERT and XLNet?

1. Table 6:
   1. Why is there an improvement when the input does not have negation? Isn't it the case that they don't do anything if the input sentence does not have negation?

**Comment by eduardoblanco** [comment]
1.i. Any languages would be fine for their purposes. To add a new language _l_, we would need:
   * a negation cue detector for _l_
   * a machine translation system from _l_ to English
   * parallel corpora (en and _l_

   I think that's all the requirements to add a new language.

**Comment by eduardoblanco** [comment]
4.i One would not be wrong, but three is better (== more empirical evidence). If it only works with one benchmark, does it really work for the "task" or that specific benchmark?

On a related note, the paper claims that the method is _model-agnostic_. That is true, but there is only empirical evidence showing that it works when coupled with RoBERTa.

**Comment by devsawgrad** [comment]
1.iii 
I think that section is needed for validation/discussion  about the quality of the dataset. 

Possible improvement in that: The manual validation of the random sampling of 100 examples for validating the quality  can be  made more robust using confidence intervals -i.e "true" accuracy is within some interval based on a significance value. 
Also the frequency of types of errors detected would  be helpful. Reporting evaluations would be helpful for comparison in future papers.

**Comment by eduardoblanco** [comment]
@devsawgrad  I agree about the frequency.

The confidence intervals would require a lot of work? Larger sample, more "humans" looking at them? It would probably be a requirement if the paper aimed at building a gold standard (which is a perfectly valid goal). If the pairs are used only to "improve NLU," you can get by demonstrating only better results with NLU.

**Reply by devsawgrad** [reply]
Fair. I see that the gold standard goal requires robust metrics.
i thought you can quantify even 1 set of samples with an interval, but then the metrics  would not be too convincing so would be better off not reporting it.

**Comment by mayanks43** [comment]
> 2.i Would it be interesting to investigate if the authors need 153k pairs to train the affirmative interpretation generator? 

I think an ablation study can help. Try training the model with smaller sets of the 153k pairs such 10k, 50k, etc. The change in performance will help understand the point of these many datapoints. 

> Would it make sense to attempt to collect more pairs?

If we see a strict increase in performance with increasing size of the datapoints, it makes sense to collect more datapoints. But if the performance increase plateaus, then there's no point. 

> 3.i What do you trust more, the automatic metrics or the manual validation?

Manual validation is more trustworthy but cannot be done for a lot of datapoints. 

> Where do the meaning of the scores come from? Do we know who did the manual validation?

The scores are labelled on a scale 0-5 on whether the given affirmative interpretation is a correct rephrasing of the sentence with negation. The authors did the manual validation on a random set of 100 generated affirmative interpretations. I don't fully understand why a 5-scale metric was needed. Perhaps, a 3-scale metric might have been enough.

**Reply by eduardoblanco** [reply]
[Nice formatting quoting the question!]

These answers are all good.

**Comment by bhavya2109sharma** [comment]
> Section 4.1
> 
> 1. What do you trust more, the automatic metrics or the manual validation?

I think both metrics are necessary for the task at hand. Automatic metrics, despite having some limitations, are crucial for offering quantitative measures and quickly assessing model performance. On the other hand, manual validation is necessary to establish a gold standard for our problem. It helps capture the nuances of the task and addresses the limitations present in automatic metrics.

> Section 4.2
>    
>    1. Why three natural language inference corpora? Can't they just use one?

Three natural language inference corpora—RTE, SNLI, and MNLI—were used in the experiment to give a wide range of data for assessing how well models perform on various entailment tasks. Since every corpus is unique, it is helpful to use a variety of corpora to evaluate how well the models generalize to different situations.

**Comment by Urvika-gola** [comment]
3.i Manual validation (to generate affirmative interpretation) has my trust, (also more interesting to do?)
A follow-up question I had in mind is when a sentence contains a negation, the sentence is sent to the Cue Detector to find its Affirmative Interpretation. After this, both - the original sentence and the affirmative interpretation is sent to the RoBERTa. I assume that the original sentence is also sent to capture the nuances of the original sentence (if they get lost in the interpretation). Could this be the only reason of doing this? What would happen if the authors used just the affirmative interpretation from the Cue Detector as the input?
